File: frontend/bhau_fitness_flutter/lib/models/class_session.dart
/// Mirrors ClassSessionDto on the backend.
class ClassSession {
  final int id;
  final int dayOfWeek; // ISO: 1=Monday .. 7=Sunday
  final String startTime; // "HH:mm:ss" as serialized by ASP.NET's TimeOnly
  final String title;
  final String trainerName;
  final String level;
  final String type;
  final int durationMin;
  final String dayLabel;
  final int capacity;
  final int bookedCount;

  ClassSession({
    required this.id,
    required this.dayOfWeek,
    required this.startTime,
    required this.title,
    required this.trainerName,
    required this.level,
    required this.type,
    required this.durationMin,
    required this.dayLabel,
    required this.capacity,
    required this.bookedCount,
  });

  /// "06:00:00" -> "06:00" for display.
  String get startTimeShort => startTime.length >= 5 ? startTime.substring(0, 5) : startTime;

  bool get isFull => bookedCount >= capacity;

  factory ClassSession.fromJson(Map<String, dynamic> json) {
    return ClassSession(
      id: json['id'] as int,
      dayOfWeek: json['dayOfWeek'] as int,
      startTime: json['startTime'] as String,
      title: json['title'] as String,
      trainerName: json['trainerName'] as String,
      level: json['level'] as String,
      type: json['type'] as String? ?? 'General',
      durationMin: json['durationMin'] as int? ?? 60,
      dayLabel: json['dayLabel'] as String? ?? '',
      capacity: json['capacity'] as int,
      bookedCount: json['bookedCount'] as int,
    );
  }
}

const weekdayNames = ['', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];

File: backend/BhauFitnessApi/Models/Entities/ClassSession.cs
using System.ComponentModel.DataAnnotations;

namespace BhauFitnessApi.Models.Entities;

/// <summary>
/// A recurring class slot on the weekly schedule (mirrors the HTML site's
/// schedule preview/booking — e.g. "Mon 6:00 AM — HIIT — Coach Aman").
/// </summary>
public class ClassSession
{
    public int Id { get; set; }

    // 1 (Monday) .. 7 (Sunday) — kept simple since classes repeat weekly.
    public int DayOfWeek { get; set; }

    public TimeOnly StartTime { get; set; }

    [Required, MaxLength(80)]
    public string Title { get; set; } = string.Empty; // "HIIT", "Yoga & Mobility", ...

    [Required, MaxLength(80)]
    public string TrainerName { get; set; } = string.Empty;

    [Required, MaxLength(20)]
    public string Level { get; set; } = "All Levels";

    // Class category, e.g. "Strength", "Cardio", "Yoga" (HTML's `type`).
    [Required, MaxLength(40)]
    public string Type { get; set; } = "General";

    public int DurationMin { get; set; } = 60;

    // Human-readable frequency label shown on the schedule ("Daily", "Mon", ...).
    [Required, MaxLength(40)]
    public string DayLabel { get; set; } = "Weekly";

    public int Capacity { get; set; } = 20;

    public bool IsActive { get; set; } = true;

    public ICollection<Booking> Bookings { get; set; } = new List<Booking>();
}
